azuread/resource_domain_service_test.go

```diff
@@ -0,0 +1,207 @@
+			return fmt.Errorf("Bad: no resource group found in state for Domain Service: %s", name)
+		}
+
+		client := testAccProvider.Meta().(*ArmClient).domainServicesClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if ar.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Domain Service %q (resource group: %s) does not exist", name, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on DomainServicesClient: %s", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureADDomainService_complete(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_network_security_group" "test" {
+  name                = "acctestNSG-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  security_rule {
+    name                        = "AllowSyncWithAzureAD"
+    priority                    = 101
+    direction                   = "Inbound"
+    access                      = "Allow"
+    protocol                    = "Tcp"
+    source_port_range           = "*"
+    destination_port_range      = "443"
+    source_address_prefix       = "AzureActiveDirectoryDomainServices"
+    destination_address_prefix  = "*"
+  }
+
+  security_rule {
+    name                        = "AllowRD"
+    priority                    = 201
+    direction                   = "Inbound"
+    access                      = "Allow"
+    protocol                    = "Tcp"
+    source_port_range           = "*"
+    destination_port_range      = "3389"
+    source_address_prefix       = "CorpNetSaw"
+    destination_address_prefix  = "*"
+  }
+
+  security_rule {
+    name                        = "AllowPSRemoting"
+    priority                    = 301
+    direction                   = "Inbound"
+    access                      = "Allow"
+    protocol                    = "Tcp"
+    source_port_range           = "*"
+    destination_port_range      = "5986"
+    source_address_prefix       = "AzureActiveDirectoryDomainServices"
+    destination_address_prefix  = "*"
+  }
+
+  security_rule {
+    name                        = "AllowLDAPS"
+    priority                    = 401
+    direction                   = "Inbound"
+    access                      = "Allow"
+    protocol                    = "Tcp"
+    source_port_range           = "*"
+    destination_port_range      = "636"
+    source_address_prefix       = "*"
+    destination_address_prefix  = "*"
+  }
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestVnet-%d"
+  address_space       = ["10.0.1.0/24"]
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  lifecycle {
+    ignore_changes = [ dns_servers ]
+  }
```

Comment:
We should not be doing this (and hence why we need to do this update in azurerm)